DOC: Describe the cookiecutter parameters

thewtex opened the pull request.

kabcode commented:
Thanks for the fast update.
Maybe a note which comments are mandatory because they affect the naming cookiecutter does? 
What if I do not want to create a filter module but a processobject based project? 

Beside those questions the doc gives enough information to fill out the promts correctly.

jhlegarreta commented on README.rst:
```diff
@@ -32,6 +32,50 @@ The following will get an external module started in a new repository::
+
 Documentation on `how to populate the module
 <https://itk.org/ITKSoftwareGuide/html/Book1/ITKSoftwareGuide-Book1ch9.html#x50-1430009>`_
 can be found in the `ITK Software Guide
```
Closing backquoute missing: `ITK Software Guide`

thewtex replied:
Done

thewtex commented:
Thanks for the review @kabcode @jhlegarreta !

> Maybe a note which comments are mandatory because they affect the naming cookiecutter does?

> I think @kabcode 's comments are fair. However, I think all of the fields described by this patch set are (or should be: e.g. full_name, email) mandatory. In that case, I do not see any need to add another note. If the comment refers to other elements (e.g. wrapping, setup.py), I'd still consider it necessary, since we tend no to do it, and this potentially reduces the target audience/adoption of the module due to a lack of Python availability.

All the parameters are mandatory, but all the parameters have defaults. I have modified the wording to try to make this more clear.

> What if I do not want to create a filter module but a processobject based project?

> As for the second comment on classes/modules based on itk::ProcessObject, may be a short comment (stating that the appropriate parent class should be chosen) on the header declaration file or this README.rst (or both) would be appropriate.

Yes, we do not have examples of all the possible classes, currently a few image to image filters, since these are the most common. I added a paragraph to explain that these classes should be removed or replaced.

> I'd even add an example folder to the module, since that would prevent accumulating too many classes for this task/keeping the ITK SW Guide up-to-date with examples.

Agreed. We now have a nice infrastructure for `examples`! A separate issue is created for this because I do not have time to address it at the moment: #34
